### PR TITLE
WT-10849 Work around edge case for mirrors with FLCS, VLCS and truncate.

### PR DESCRIPTION
**Pull request by sueloverso**
The VLCS characteristic of appending to the end of a table means that a truncate at the end of a table and then an append-insert can overwrite the mirror space. It is correctly handled except when both FLCS and VLCS exist as mirrors. In that case I adjust the truncate range to leave the last mirror element in the table.

### CODE REVIEW

File: test/format/ops.c [hunk 1]
```diff
@@ -1163,6 +1163,16 @@ ops(void *arg)
                     tinfo->last += range;
                     if (tinfo->last > max_rows)
                         tinfo->last = 0;
+                    /*
+                     * Edge case: If we are truncating to the end of the table and we're doing
+                     * mirroring with both a VLCS and FLCS table, then we can get a false mismatch
+                     * failure if the truncate happens before the first append-only insert to the
+                     * VLCS table. So if we have VLCS and FLCS mirrors, set last to the second last
+                     * key.
```
Comment (keitharnoldsmith) [inline test/format/ops.c:1171]: I found the explanation of this edge case on the ticket easier to understand than the comment here. So maybe reword or copy the other text?

I think the difference is that the ticket spells out the problematic sequence of events with more detail.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM. I left a comment about a comment :-)

It seems a bit inelegant to special case this one edge case, but I can't think of anything better. And has the benefit of only reducing what we test in that specific case.

File: test/format/ops.c (hunk 1 above)
Comment (sueloverso) [reply test/format/ops.c]: I pulled that in an got rid of the specific key numbers. Please reread.

File: test/format/ops.c [hunk 2]
```diff
@@ -1164,11 +1164,23 @@ ops(void *arg)
                     if (tinfo->last > max_rows)
                         tinfo->last = 0;
                     /*
-                     * Edge case: If we are truncating to the end of the table and we're doing
-                     * mirroring with both a VLCS and FLCS table, then we can get a false mismatch
-                     * failure if the truncate happens before the first append-only insert to the
-                     * VLCS table. So if we have VLCS and FLCS mirrors, set last to the second last
-                     * key.
+                     * Edge case: There is a case where we cannot detect a proper mirror mismatch.
+                     * Say we truncated the tail end key range of all the mirrors from N-max_rows.
```
Comment (keitharnoldsmith) [inline test/format/ops.c:1168]: I'd change `N-max_rows` to something like `N to max_rows` to avoid possible confusion the hyphen is meant as a minus sign.

File: test/format/ops.c [hunk 3]
```diff
@@ -1164,11 +1164,23 @@ ops(void *arg)
...
+                     * This truncate happened before any thread added another non-mirrored
+                     * append/insert to a VLCS table and the data in that truncated key range was
+                     * sufficient to delete the pages at the end of the VLCS table. Then when a VLCS
+                     * non-mirrored insert happened, it appended the new item at key N instead of at
+                     * max_rows + 1.
+                     *
+                     * Then the next mirror check hit will mismatch if there is an FLCS table
```
Comment (keitharnoldsmith) [inline test/format/ops.c:1175]: Cut `hit`

**Review by keitharnoldsmith — APPROVED** [review]
LGTM, with a couple of proof-reading level comments.
